#!/usr/bin/env R

# Import the dataframe from each bam file

# Merge each dataframe into a single object.

# Count the number of concordant alignments vs the number of discordant alignments.

# Create a histogram of the alignment lengths

# Create a 2d plot of the error rate against the alignment length.


# There are many more discordant alignments. Why is this so?


# Plot the supplementary vs primary across the genome.